test/from-draft3/to-draft4/type-has-atleast-one-object-in-its-array-anyOf-absent.json

```diff
@@ -0,0 +1,12 @@
+[
+    {
+        "title": "`type` is present and contains atleast one object in it",
+        "from": {
+            "type": [ "string", { "type": "string" } ]
+        },
+        "to": {
+            "type": [ "string", { "type": "string" } ],
```

Comment:
Draft 4 doesn't allow schemas in `type`

Reply:
I have included it as a pre processing step, so that we can push to anyOf in the net rule, OR if schema should be valid at any point of transformation as you said before, we have to change `{} - match-all` saying that use single {} in conditions and single {} in transform, and series of transform will be like:
1. add anyOf - will just add once
2. remove-and-append type, {} - will juts remoe all matchinng and append all
3. remove type if empty - will remove type

So based on the series the transform will take place...

Reply:
I'm not sure I follow but it might be the same problem we discussed in another PR: tests are not expected to exercise a single rule. I think what you call a "pre processing step" is an intermediate state between rules, right? If so, we shouldn't be testing the intermediate stakes, only the before and after of the ENTIRE set of rules in one shot

Reply:
This is resolved by considering tests as whole and not for single rule, so I'll resolve this comment

Reply:
Cool. Let me know when its ready to review again